File: 1로 만들기 .cpp
#include<iostream>
using namespace std;
long long int n;
long long int arr[1000001] = { 0, };
int main() {
	cin >> n;
	for (int i = 2; i < n + 1; i++) {
		arr[i] = arr[i - 1] + 1;
		if (i % 3 == 0) arr[i] = min(arr[i], arr[i / 3] + 1);
		if (i % 2 == 0) arr[i] = min(arr[i], arr[i / 2] + 1);
	}
	cout << arr[n];
}

File: 1로 만들기.py
n = int(input())
arr = [987654321] * (n+1)
arr[1] = -1
for i in range(1, n+1):
    if i % 3 == 0:
        arr[i] = arr[int(i/3)]
    if i % 2 == 0:
        arr[i] = min(arr[int(i/2)], arr[i]) 
    arr[i] = min(arr[i],arr[i-1]) + 1
print(arr[n])